Description: Fix compilation error when compiling with gcc 10
Author: Matthijs Kooijman <matthijs@stdin.nl>
Forwarded: https://github.com/OpenTTD/grfcodec/pull/6
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=957307
Bug: https://github.com/OpenTTD/grfcodec/issues/5
--- a/src/command.cpp
+++ b/src/command.cpp
@@ -300,7 +300,7 @@ bool parse_comment(const string&line){
 		break;
 	case BEAUTIFY:{
 		commandstream>>command_part;
-		uint val=find_command(command_part,beaut),togglebit;
+		int val=find_command(command_part,beaut),togglebit;
 		if(val!=(uint)-1&&val!=OFF)_commandState.beautifier=true;
 		switch(val){
 		case -1:
